### PR TITLE
feat: update provider version and fix errors

### PR DESCRIPTION
**Pull request by ChapChap**

### CODE REVIEW

File: examples/cluster_with_ip_addresses/main.tf [hunk 1]
```diff
@@ -25,9 +25,9 @@ module "custom_service_account" {
 }
 
 module "custom_network" {
-  source = "git@github.com:padok-team/terraform-google-network.git"
-  project = "<YOUR_PROJECT_ID>"
-  name = "my-super-duper-cluster-network"
+  source  = "git@github.com:padok-team/terraform-google-network.git"
+  project = "padok-cloud-factory"
```
Comment (ChapChap) [inline examples/cluster_with_ip_addresses/main.tf:29]:
```suggestion
  project = "<YOUR_PROJECT_ID>"
```

**Comment by busser** [comment]
@Kimelyne The version constraints should be of the form `~> 4.12` rather than `>= 4.12.0`, so as to not allow using version 5.x of the provider, once it comes out.

File: examples/cluster_with_ip_addresses/main.tf [hunk 2]
```diff
@@ -1,14 +1,14 @@
 # Short description of the use case in comments
 
 provider "google" {
-  project = "<YOUR_PROJECT_ID>"
+  project = "padok-cloud-factory"
   region  = "europe-west1"
 }
 
 module "custom_service_account" {
   source = "git@github.com:padok-team/terraform-google-serviceaccount.git"
```
Comment (ChapChap) [inline examples/cluster_with_ip_addresses/main.tf:9]:
```suggestion
  source = "git@github.com:padok-team/terraform-google-serviceaccount.git?ref=v1.0.0"
```

File: examples/cluster_with_ip_addresses/main.tf [hunk 3]
```diff
@@ -25,9 +25,9 @@ module "custom_service_account" {
 }
 
 module "custom_network" {
-  source = "git@github.com:padok-team/terraform-google-network.git"
-  project = "<YOUR_PROJECT_ID>"
-  name = "my-super-duper-cluster-network"
+  source  = "git@github.com:padok-team/terraform-google-network.git"
```
Comment (ChapChap) [inline examples/cluster_with_ip_addresses/main.tf:28]:
```suggestion
  source  = "git@github.com:padok-team/terraform-google-network.git?ref=v2.0.3"
```

File: examples/regional_private_cluster/main.tf [hunk 4]
```diff
@@ -25,9 +25,9 @@ module "custom_service_account" {
 }
 
 module "custom_network" {
-  source = "git@github.com:padok-team/terraform-google-network.git"
+  source  = "git@github.com:padok-team/terraform-google-network.git"
```
Comment (ChapChap) [inline examples/regional_private_cluster/main.tf:28]:
```suggestion
  source  = "git@github.com:padok-team/terraform-google-network.git?ref=v2.0.3"
```

**Comment by elilette** [comment]
Validation par Alan pad premier

**Pull request closed without merging**